pyfr/solvers/euler/elements.py

```diff
@@ -100,3 +104,49 @@ def set_backend(self, *args, **kwargs):
                 u=s(self._scal_qpts, l), f=s(self._vect_qpts, l),
                 verts=self.ploc_at('linspts', l), upts=self.qpts
             )
+
+        if self.cfg.get('solver', 'shock-capturing') == 'entropy-filter':
+            # Minimum density/pressure constraints
+            d_min = float(self.cfg.get('solver-entropy-filter', 'd_min', 1e-6))
+            p_min = float(self.cfg.get('solver-entropy-filter', 'p_min', 1e-6))
+            # Absolute entropy tolerance
+            e_atol = float(self.cfg.get('solver-entropy-filter', 'e_atol', 1e-6))
+            # Relative entropy tolerance (with respect to maximum variation in entropy within element)
+            e_rtol = float(self.cfg.get('solver-entropy-filter', 'e_rtol', 1e-3))
+            # Number of iterations to compute filter strength
+            niters = int(self.cfg.get('solver-entropy-filter', 'niters', 10))
+            # Maximum filter strength
+            precision = self.cfg.get('backend', 'precision')
+            zeta_max = -np.log(1e-7) if precision == 'single' else -np.log(1e-16)
```

Comment:
Use the Python log functions rather than the NumPy ones (which return a custom scalar type)

Reply:
Can also ask for `eps` from backend.fpdtype